common/kvstore/leveldb/leveldb_store.go

```diff
@@ -16,27 +18,40 @@ var _ kvstore.Store[[]byte] = &levelDBStore{}
 
 // levelDBStore implements kvstore.Store interfaces with levelDB as the backend engine.
 type levelDBStore struct {
-	db   *leveldb.DB
-	path string
-
-	logger logging.Logger
-
+	db       *leveldb.DB
```

Comment:
consider adding an inline `sync.Mutex` to make this thread safe

Reply:
https://github.com/Layr-Labs/eigenda/pull/1313/commits/a55f370def572ac584a0564747a6cb8f834cc1f5